react-components/src/tree-view/view/components/tree-view-load-more.tsx

```
@@ -1,40 +1,44 @@
-/* eslint-disable react/prop-types */
 import { type ReactElement } from 'react';
-
 import { Button } from '@cognite/cogs.js';
-
 import { type TreeNodeType } from '../../model/tree-node-type';
-import { type TreeNodeAction } from '../../model/types';
-import { type AdvancedTreeViewProps } from '../advanced-tree-view-props';
 import { HORIZONTAL_SPACING, INDENTATION, LOAD_MORE_LABEL } from '../constants';
+import { type ILazyLoader } from '../../model/i-lazy-loader';
+
+export type TreeViewLoadMoreProps = {
+  node: TreeNodeType;
+  level: number;
+  label?: string;
+  loader: ILazyLoader;
+};
 
 export const TreeViewLoadMore = ({
   node,
-  onClick,
   level,
-  props
-}: {
-  node: TreeNodeType;
-  onClick: TreeNodeAction;
-  level: number;
-  props: AdvancedTreeViewProps;
-}): ReactElement => {
+  label,
+  loader
+}: TreeViewLoadMoreProps): ReactElement => {
   const horizontalSpacing = HORIZONTAL_SPACING / 2 + 'px';
   const marginLeft = (level + 1) * INDENTATION + 'px';
   return (
     <Button
       type="secondary"
       size="small"
+      aria-label={label}
       style={{
         gap: horizontalSpacing,
         padding: '4px',
         marginTop: horizontalSpacing,
         marginLeft
       }}
-      onClick={() => {
-        onClick(node);
+      onClick={async () => {
+        if (node.loadSiblings === undefined) return;
+        try {
+          await node.loadSiblings(loader);
+        } catch (error) {
+          console.error('Failed to load more nodes:', error);
```

Why do we swallow the error here? I also am not a fan of console.error here, since neither the user, nor we will see this. There isn't any real value in console logging this. I would either remove the try / catch completly and then let the error bubble to global scope, or, if we actually don't care about this erroring, do nothing in the catch and just continue execution as if it was successful.

---

I wasn't sure about this. I have not find any rules in the code how we should threat. I have remove it the try block.